llama_finetuning.py

```diff
@@ -83,25 +86,48 @@ def main(**kwargs):
         world_size = int(os.environ["WORLD_SIZE"])
 
     if torch.distributed.is_initialized():
-        torch.cuda.set_device(rank)
+        torch.cuda.set_device(local_rank)
```

Comment:
What is the reason for replacing "rank" with "local_rank"?

Reply:
@chauhang I think the original usage of rank was a bug, maybe from copy-paste error or typo.  This function expects local rank so I fixed it.

This has been validated in two different ways:
1. I have validated running in both single node and multi node env, and can confirm the fix works but previous didn't work in multi-node env.  Same observation from others as discussed in https://github.com/facebookresearch/llama-recipes/issues/90
2. I also double checked other repos from pytorch and confirm the correctness and consistency of using local_rank, e.g. https://github.com/pytorch/workshops/blob/master/FSDP_Workshop/main_training.py#L382

Reply:
we should be using local_rank here.  Rank is really the global rank.  
Local_rank = for intra-node (within node) assignments of gpu ids.

For single node, rank is the same numerically as local_rank and they are equivalent ids and will thus work. 

But for multi-node, rank is global_rank, unique across all nodes,  and will result in a failing gpu id within the second+ nodes if used in gpu device assignment since it would be trying to assign gpu ids higher than what the local node has.